gcs_storage_client.go

```diff
@@ -83,8 +89,6 @@ func (util *snowflakeGcsClient) getFileHeader(meta *fileMetadata, filename strin
 				meta.lastError = fmt.Errorf(resp.Status)
 				meta.resStatus = renewToken
 			}
-			meta.lastError = fmt.Errorf(resp.Status)
-			meta.resStatus = errStatus
```

Comment:
changing position of this lines, could cause some behavior change as values were overwritten before and now it will see retry errors. Is that intentional?

Reply:
It's intentional. Previously the error status was always overwritten to errStatus, so for example, if the status code was 404, the status notFoundFile would be overwritten to errStatus at the end of the block. If we set the errStatus first, it won't override other statuses and will remain as the default.